django/db/models/functions/text.py

```diff
@@ -65,83 +65,51 @@ def as_sqlite(self, compiler, connection, **extra_context):
         return super().as_sql(compiler, connection, function="CHAR", **extra_context)
 
 
-class ConcatPair(Func):
+class Concat(Func):
     """
-    Concatenate two arguments together. This is used by `Concat` because not
-    all backend databases support more than two arguments.
+    Concatenate expressions together. Expressions that are nullable
+    are coalesced to an empty string and non-textual expressions are
+    casted to strings.
     """
 
-    function = "CONCAT"
+    function = None
+    template = "%(expressions)s"
+    arg_joiner = " || "
```

Comment:
I used a custom function exactly like this one, to be able to test GeneratedField with strings concatenation in PostgreSQL. 🙂